src/Microsoft.Diagnostics.Monitoring.StartupHook/Exceptions/ExceptionIdentifier.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Reflection;
+
+namespace Microsoft.Diagnostics.Monitoring.StartupHook.Exceptions
+{
+    /// <summary>
+    /// Class representing a type of exception thrown from the specific location
+    /// in code. This class may be used as an identifier for determining whether
+    /// exception usages are equivalent (same exception type thrown from the same location).
+    /// </summary>
+    internal sealed record class ExceptionIdentifier
+    {
+        public ExceptionIdentifier(Exception ex)
+        {
+            ArgumentNullException.ThrowIfNull(ex);
+
+            ExceptionType = ex.GetType();
+
+            StackTrace stackTrace = new(ex, fNeedFileInfo: false);
```

Comment:
I'm assuming that there's minimal overhead here compared to calling Exception.StackTrace correct?

Reply:
Correct, turning it off avoids reading in source line information and processors.

Reply:
`Exception.StackTrace` effectively calls 'new StackTrace(this, fNeedFileInfo: true)`, and we don't need the source line information. Also, `Exception.StackTrace` is a string and I don't want to parse that.